internal/endpoints/rulesfn/doc.go

```diff
@@ -0,0 +1,4 @@
+// Package rulesfn provides endpoint rule functions for evaluating endpoint
+// resolution rules.
+
+package rulesfn
```

Comment:
**Correctness**: Not sure where to place this comment but having rule functions in an `internal` package won't work. These functions will need to be referenced eventually by generated service clients which means they need to be accessible outside of the `smithy-go` package.

Reply:
ah right, ok will make this public. that actually might work out fine with a top-level `endpoints` package and i can put the `error_collector.go` file in there (rather than introducing a top-level `errors` package)